src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h

```diff
@@ -0,0 +1,745 @@
+
+public:
+    /**
+     * Constructor
+     * @param kinDyn KinDynComputations object
+     * @param handler variable handler object
+     * @param type type of control (position/orientation/pose)
+     * @param frameName name of the frame that has to be controlled (if CoM the CoM will be
+     * controlled)
+     * @param axisName X, Y or Z coordinate that will be controlled
+     * @throw std::runtime_error in case of an undefined variable/frame
+     */
+    CartesianElement(iDynTree::KinDynComputations& kinDyn,
+                     const VariableHandler& handler,
+                     const Type& type,
+                     const std::string& frameName,
+                     const AxisName& axisName = AxisName::X);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration desired linear acceleration
+     * @param velocity desired linear velocity
+     * @param position desired position
+     * @throw std::runtime_error if the Type is different from POSE and POSITION
+     */
+    void setDesiredTrajectory(const iDynTree::Vector3& acceleration,
+                              const iDynTree::Vector3& velocity,
+                              const iDynTree::Vector3& position);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration  desired angular acceleration
+     * @param velocity desired  angular velocity
+     * @param rotation desired rotation matrix
+     * @throw std::runtime_error if the Type is different from POSE and ORIENTATION
+     */
+    void setDesiredTrajectory(const iDynTree::Vector3& acceleration,
+                              const iDynTree::Vector3& velocity,
+                              const iDynTree::Rotation& rotation);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration desired spatial acceleration
+     * @param velocity angular desired spatial velocity
+     * @param transform desired homogeneous transform
+     * @throw std::runtime_error if the Type is different from POSE
+     */
+    void setDesiredTrajectory(const iDynTree::SpatialAcc& acceleration,
+                              const iDynTree::Twist& velocity,
+                              const iDynTree::Transform& transform);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration desired acceleration
+     * @param velocity angular desired velocity
+     * @param transform desired position
+     * @throw std::runtime_error if the Type is different from ONE_DIMENSION
+     */
+    void setDesiredTrajectory(const double& acceleration,
+                              const double& velocity,
+                              const double& position);
+
+    /**
+     * Set the linear PID gains
+     * @param kp proportional gain
+     * @param kd derivative gain
+     * @throw std::runtime_error if the Type is different from POSE and POSITION
+     */
+    void setLinearPIDGains(const iDynTree::Vector3& kp, const iDynTree::Vector3& kd);
```

Comment:
It is called ``PID`` but you can set only P and D.

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/3/commits/906d16a11bb246ca66d4fb00ff35957cae812e33